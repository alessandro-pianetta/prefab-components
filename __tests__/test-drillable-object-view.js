import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import DrillableObjectView from '../src/index';


it('renders correctly with no props', () => {
  const drillableObjectView = renderer.create(<DrillableObjectView />).toJSON();

  expect(drillableObjectView).toMatchSnapshot();
});

it('renders correctly with a left margin as a prop', () => {
  const drillableObjectView = renderer.create(<DrillableObjectView keyName="aKeyName" marginLeft={10} />).toJSON();

  expect(drillableObjectView).toMatchSnapshot();
});

it('renders correctly with a autoExpandDepth as a prop', () => {
  const drillableObjectView = renderer.create(<DrillableObjectView keyName="aKeyName" autoExpandDepth={5} />).toJSON();

  expect(drillableObjectView).toMatchSnapshot();
});

it('renders correctly with an array value', () => {
  const drillableObjectView = renderer.create(<DrillableObjectView keyName="arrayKey" value={[1, 2, 3]} />).toJSON();

  expect(drillableObjectView).toMatchSnapshot();
});

it('renders correctly with a boolean value', () => {
  const drillableObjectView = renderer.create(<DrillableObjectView keyName="booleanKey" value />).toJSON();

  expect(drillableObjectView).toMatchSnapshot();

  const falseDrillableObjectView = renderer.create(<DrillableObjectView keyName="booleanKey" value={false} />).toJSON();

  expect(falseDrillableObjectView).toMatchSnapshot();
});

it('renders correctly with a null value', () => {
  const drillableObjectView = renderer.create(<DrillableObjectView keyName="nullKey" value={null} />).toJSON();

  expect(drillableObjectView).toMatchSnapshot();
});

it('renders correctly with a boolean value', () => {
  const drillableObjectView = renderer.create(<DrillableObjectView keyName="objectKey" value={{ subObjectKey: {} }} />).toJSON();

  expect(drillableObjectView).toMatchSnapshot();
});

it('renders correctly with a boolean value', () => {
  const drillableObjectView = renderer.create(<DrillableObjectView keyName="stringKey" value="stringValue" />).toJSON();

  expect(drillableObjectView).toMatchSnapshot();
});
