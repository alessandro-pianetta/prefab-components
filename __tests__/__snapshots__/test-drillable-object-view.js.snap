// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with a autoExpandDepth as a prop 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Object {
      "marginLeft": 8,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#313A42",
      }
    }
  >
    aKeyName
    :
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#313A42",
      }
    }
  >
     
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#818181",
        }
      }
    >
      undefined
    </Text>
  </Text>
</Text>
`;

exports[`renders correctly with a boolean value 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Object {
      "marginLeft": 8,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#313A42",
      }
    }
  >
    booleanKey
    :
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#313A42",
      }
    }
  >
     
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#1B00D1",
        }
      }
    >
      true
    </Text>
  </Text>
</Text>
`;

exports[`renders correctly with a boolean value 2`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Object {
      "marginLeft": 8,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#313A42",
      }
    }
  >
    booleanKey
    :
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#313A42",
      }
    }
  >
     
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#1B00D1",
        }
      }
    >
      false
    </Text>
  </Text>
</Text>
`;

exports[`renders correctly with a boolean value 3`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "marginLeft": 8,
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#313A42",
      }
    }
  >
    objectKey
    : +
  </Text>
</View>
`;

exports[`renders correctly with a boolean value 4`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Object {
      "marginLeft": 8,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#313A42",
      }
    }
  >
    stringKey
    :
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#313A42",
      }
    }
  >
     
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#C41A17",
        }
      }
    >
      "
      stringValue
      "
    </Text>
  </Text>
</Text>
`;

exports[`renders correctly with a left margin as a prop 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Object {
      "marginLeft": 10,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#313A42",
      }
    }
  >
    aKeyName
    :
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#313A42",
      }
    }
  >
     
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#818181",
        }
      }
    >
      undefined
    </Text>
  </Text>
</Text>
`;

exports[`renders correctly with a null value 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Object {
      "marginLeft": 8,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#313A42",
      }
    }
  >
    nullKey
    :
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#313A42",
      }
    }
  >
     
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#818181",
        }
      }
    >
      null
    </Text>
  </Text>
</Text>
`;

exports[`renders correctly with an array value 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "marginLeft": 8,
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#313A42",
      }
    }
  >
    arrayKey
    : +
  </Text>
</View>
`;

exports[`renders correctly with no props 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Object {
      "marginLeft": 8,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#313A42",
      }
    }
  >
    parent
    :
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#313A42",
      }
    }
  >
     
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#818181",
        }
      }
    >
      undefined
    </Text>
  </Text>
</Text>
`;
